### DISCUSSION TITLE
Form Customizer Feedback

### DISCUSSION THREAD

**Discussion by AJIXuMuK** [SharePoint/sp-dev-docs#8337] [Ideas]
In SPFx 1.15.0 we released new extension type: Form Customizer.
We've started with some basic implementation and want to hear your feedback on what else is needed to simplify development of the form customizers.
I will start with few things and please feel free to add your ideas.

### Form Customizer in Panel
Currently form customizer support full page experience. And we know we need to make it work in panel, similarly to OOB forms.

### List Item Data
We will provide list item data as part of the context to allow devs avoid addition REST call to retrieve this data.

### JSON Form Formatting as part of the context
We've got feedback that it would be nice to have information about JSON formatting applied to the form. And we will include it.

Let's discuss what else is needed. We will really appreciate if you also share use cases.

Thanks!

**Comment by DonKirkham** [comment]
I came here to put in a request to have it available in a panel similar to OOTB and PowerApp, but you already have it in your known requests. so I'll just +1.

**Comment by siddharth-vaghasia** [comment]
Hey @AJIXuMuK - This is more of quick question or confirmation but if this is not already added to list it can be good feature request..

So for another type of extension(not form extension), when we create SPFx solution it automatically create element.xml and ClientSideInstance inside SharePoint which allows us to use Feature framework to associate the extension....as of last releasae for form extension these files are not created automatically? Is this in the roadmap or feature request already?

**Reply by AJIXuMuK** [reply]
@siddharth-vaghasia - great feedback, thanks!
No, the files are not generated.
The reason for that is the form customizer is applied on content type level.
We can add `elements.xml` to provision custom content type and demo list using this content type.
But the question here if that is a common scenario or no.

**Reply by siddharth-vaghasia** [reply]
As of now what we have is that, if we build a form customizer we would need to use REST API or CSOM to associate this form customizer with the List's content type...It was more like we don't have to do an extra step to associate a form Customizer with List but it can use a feature framework to associate with existing one or more content types... 

This can also be useful when we want to associate the same SPFx form customizer with multiple lists... One of my customers has a provisioning engine where they are providing 100 of sites but they have some custom logic on the list form using Infopath and they wanted to get rid of Infopath... We cannot associate PowerApps List form because then to import/export a PowerApp List form we have to do it manually... So I was proposing Form customizer which can be associated with content type when SPFx solution is installed which create list and also associate the form customizer with it via content type..

**Reply by AJIXuMuK** [reply]
Yeah, if you want to create a list and a content type during app installation - we can definitely add.

**Comment by andrewconnell** [comment]
Should have an option for a site owner/designer to edit the settings on the content type/list for the custom forms, ideally selecting the installed form customizers. Having to write code to install these 100% of the time is not ideal.

However, it is nice to have an API deployment option as a 1st step... so we can install them ourselves. 

What would be REALLY nice is if we had events when apps were installed/uninstalled, like webhook, so we could automatically register/unregister them.

**Comment by andrewconnell** [comment]
When showing the form in the full screen mode, would be nice to have context for the current list included on the top-right of the form... didn't classic forms work this way originally so users had context?
